bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/handler/RotelHandler.java

```diff
@@ -688,180 +680,108 @@ public void handleCommand(ChannelUID channelUID, Command command) {
-                                    RotelCommand.ZONE2_MUTE_OFF, RotelCommand.ZONE2_MUTE_TOGGLE);
-                        } else {
-                            success = false;
-                            logger.debug("Command {} from channel {} failed: unavailable feature", command, channel);
-                        }
-                        break;
                     case CHANNEL_ZONE3_MUTE:
-                        if (!powerZone3) {
-                            success = false;
-                            logger.debug("Command {} from channel {} ignored: zone 3 in standby", command, channel);
-                        } else if (model.hasVolumeControl() && model.hasZone3Commands()) {
-                            handleMuteCmd(false, channel, command, RotelCommand.ZONE3_MUTE_ON,
-                                    RotelCommand.ZONE3_MUTE_OFF, RotelCommand.ZONE3_MUTE_TOGGLE);
-                        } else {
-                            success = false;
-                            logger.debug("Command {} from channel {} failed: unavailable feature", command, channel);
-                        }
-                        break;
                     case CHANNEL_ZONE4_MUTE:
-                        if (!powerZone4) {
+                        if (numZone == 0 && !isPowerOn()) {
                             success = false;
-                            logger.debug("Command {} from channel {} ignored: zone 4 in standby", command, channel);
-                        } else if (model.hasVolumeControl() && model.hasZone4Commands()) {
-                            handleMuteCmd(false, channel, command, RotelCommand.ZONE4_MUTE_ON,
-                                    RotelCommand.ZONE4_MUTE_OFF, RotelCommand.ZONE4_MUTE_TOGGLE);
+                            logger.debug("Command {} from channel {} ignored: device in standby", command, channel);
+                        } else if (numZone > 0 && !isZonePowerOn(numZone)) {
+                            success = false;
+                            logger.debug("Command {} from channel {} ignored: zone {} in standby", command, channel,
+                                    numZone);
+                        } else if (model.hasVolumeControl() && (numZone == 0 || model.hasZoneCommands(numZone))) {
+                            handleMuteCmd(numZone == 0 && protocol == RotelProtocol.HEX, channel, command,
+                                    getMuteOnCommand(numZone), getMuteOffCommand(numZone),
+                                    getMuteToggleCommand(numZone));
                         } else {
                             success = false;
                             logger.debug("Command {} from channel {} failed: unavailable feature", command, channel);
                         }
                         break;
                     case CHANNEL_BASS:
                     case CHANNEL_MAIN_BASS:
-                        if (!isPowerOn()) {
+                    case CHANNEL_ZONE1_BASS:
```

Comment:
I'm a little confused as it seems concept has changed from "additional number of zones" to "total number of zones including main zone" (according to `RotelModel`). So when having only main zone, shouldn't numZone be 1, not 0? And is `CHANNEL_MAIN_BASS` representing the same zone as `CHANNEL_ZONE1_BASS` or is the latter in fact the first additional zone?

Reply:
I need to analyze more in details. But at least, it needs to be clarified.

Reply:
In fact main zone and zone 1 are not exactly the same thing. I removed mention of "main zone" in `RotelModel`.
Only in `RotelSource`, I mix together the associated main zone command and the zone 1 command, because in practice a model cannot have main zone commands and zone 1 commands.
So I prefer keeping the code as it is, I just simplify a little with the introduction of the method isPowerOn(int numZone).